Properly escape double-quotes inside commands

Phrogz opened the pull request:
Fixes #62

archSeer commented on lib/ruby-mpd/parser.rb:
```diff
@@ -24,15 +24,15 @@ def convert_command(command, *params)
           end
         when MPD::Song
-          %Q["#{param.file}"] # escape filename
+          %Q["#{param.file.gsub('"','\\"')}"] # escape filename
```
What about using http://ruby-doc.org/stdlib-1.9.3/libdoc/shellwords/rdoc/Shellwords.html#method-c-escape instead?

Phrogz replied:
`Shellwords.escape` does not seem quite appropriate to me. From the documentation there:

> _Note that a resulted string should be used unquoted and is not intended for use in double quotes nor in single quotes._

MPD requires us to wrap in double-quotes; you can't just backslash escape characters.

Filename    | MPD requires    | Shellwords creates
------------|-----------------|-------------------
`foo.mp3`   | `foo.mp3`       | `foo.mp3`
`a b.mp3`   | `"a b.mp3"`     | `a\ b.mp3`
`bob's.mp3` | `"bob's.mp3"`   | `bob\'s.mp3`
`"yay".mp3` | `"\"yay\".mp3"` | `\"yay\".mp3`

We _could_ wrap the result of `Shellwords.escape` in double quotes and the results would be fine. However, all that does is add superfluous backslashes by calling `gsub` with a larger set of characters.

Would you prefer that I extract all escaping into a re-used utility method that is hand-tailored to MPD's needs? e.g.

```ruby
def mpd_argument(str)
   str =~ /['"\s]/ ? "#{str.gsub '"','\\"'}" : str
end
```

That way, if we find more bugs, more characters that need escaping, the fix is DRY.

archSeer replied:
Yeah, I think extracting it in a utility function would be good, so that we can update it easily in the future, besides that LGTM :+1:

Phrogz commented:
Hey! The specs actually caught a bug in my code. How delightful :)

I've fixed the bug, and also updated the specs to more accurately match MPD's requirements (instead of the previous implementation).

archSeer commented on lib/ruby-mpd/parser.rb:
```diff
@@ -25,18 +25,24 @@ def convert_command(command, *params)
+
+    # MPD requires that certain parameters be double-quoted
+    def quotable_param(value)
+      unless value.nil?
```
minor: `return if value.nil?`, that way you get rid of the extra level of indentation

archSeer commented:
Looking good :+1: pending the minor comment :)

Phrogz commented:
I'm personally in favor of coding using a more functional style than imperative, but this is your project, so not a problem. I've de-nested that method.
